src/tdom/component_signature.py

```diff
@@ -0,0 +1,22 @@
+"""Inspect functions and dataclasses to match props to args."""
+
+from dataclasses import is_dataclass, fields
+from inspect import signature
```

Comment:
this breaks in MicroPython

Reply:
Right, I kind of structured these changes to show that code here might ship in a different package, and people could `html()` from the other package if they are 100% SSR and no MicroPython.

Reply:
MicroPython has a server too, hence it could be 100% SSR **on** MicroPython ... I think one package that can offer more where more is possible is not a bad idea but the core needs to be settled.

Reply:
some link:

  * https://randomnerdtutorials.com/esp32-esp8266-micropython-web-server/
  * https://github.com/troublegum/micropyserver
  * https://docs.micropython.org/en/latest/esp8266/tutorial/network_tcp.html